extensions/amp-a4a/0.1/real-time-config-manager.js

```diff
@@ -0,0 +1,163 @@
+import {RTC_VENDORS} from './callout-vendors.js';
+import {tryParseJson} from '../../../src/json';
+import {dev, user} from '../../../src/log';
+import {Services} from '../../../src/services';
+import {isArray, isObject} from '../../../src/types';
+import {isSecureUrl} from '../../../src/url';
+
+/** @type {string} */
+const TAG = 'real-time-config';
+
+/** @type {number} */
+export const MAX_RTC_CALLOUTS = 5;
+
+/**
+ * For a given A4A Element, sends out Real Time Config requests to
+ * any urls or vendors specified by the publisher.
+ * @param {!AMP.BaseElement} a4aElement
+ * @param {!Object} customMacros The ad-network specified macro
+ *   substitutions available to use.
+ */
+function realTimeConfigManager(a4aElement, customMacros) {
+  const rtcConfig = validateRtcConfig(a4aElement.element);
+  if (!rtcConfig) {
+    return;
+  }
+  const promiseArray = [];
+  const seenUrls = {};
+  const rtcStartTime = Date.now();
+  // For each publisher defined URL, inflate the url using the macros,
+  // and send the RTC request.
+  (rtcConfig['urls'] || []).forEach(url => {
+    inflateAndSendRtc_(a4aElement, url, seenUrls, promiseArray,
+        rtcStartTime, customMacros,
+        rtcConfig['timeoutMillis']);
+  });
+  // For each vendor the publisher has specified, inflate the vendor
+  // url if it exists, and send the RTC request.
+  Object.keys(rtcConfig['vendors'] || []).forEach(vendor => {
+    const url = RTC_VENDORS[vendor.toLowerCase()];
+    if (!url) {
+      user().warn(TAG, `unknown vendor ${vendor}`);
```

Comment:
we should pass these errors along in the promise array so that the impl gets reference to ALL specified urls/vendors